packages/ERTP/src/typeGuards.js

```diff
@@ -114,18 +114,120 @@ harden(isCopyBagValue);
 // One GOOGOLth should be enough decimal places for anybody.
 export const MAX_ABSOLUTE_DECIMAL_PLACES = 100;
 
-export const AssetValueShape = M.or('nat', 'set', 'copySet', 'copyBag');
+export const AssetKindShape = M.or('nat', 'set', 'copySet', 'copyBag');
 
 export const DisplayInfoShape = M.partial(
   harden({
     decimalPlaces: M.and(
       M.gte(-MAX_ABSOLUTE_DECIMAL_PLACES),
       M.lte(MAX_ABSOLUTE_DECIMAL_PLACES),
     ),
-    assetKind: AssetValueShape,
+    assetKind: AssetKindShape,
   }),
   harden({
     // Including this empty `rest` ensures that there are no other
     // properties beyond those in the `base` record.
   }),
 );
+
+// //////////////////////// Interfaces /////////////////////////////////////////
+
+export const BrandShape = M.remotable('Brand');
+export const IssuerShape = M.remotable('Issuer');
+export const PaymentShape = M.remotable('Payment');
+export const PurseShape = M.remotable('Purse');
+export const DepositFacetShape = M.remotable('DepositFacet');
+const NotifierShape = M.remotable('Notifier');
+export const MintShape = M.remotable('Mint');
+
+export const BrandI = M.interface('Brand', {
+  isMyIssuer: M.callWhen(M.await(IssuerShape)).returns(M.boolean()),
+  getAllegedName: M.call().returns(M.string()),
+  getDisplayInfo: M.call().returns(DisplayInfoShape),
+  getAmountShape: M.call().returns(M.pattern()),
+});
+
+/**
+ * @param {Pattern} [brandShape]
+ * @param {Pattern} [assetKindShape]
+ * @param {Pattern} [amountShape]
+ */
+export const makeIssuerInterfaces = (
+  brandShape = BrandShape,
+  assetKindShape = AssetKindShape,
+  amountShape = AmountShape,
+) => {
+  const IssuerI = M.interface('Issuer', {
+    getBrand: M.call().returns(brandShape),
+    getAllegedName: M.call().returns(M.string()),
+    getAssetKind: M.call().returns(assetKindShape),
+    getDisplayInfo: M.call().returns(DisplayInfoShape),
+    makeEmptyPurse: M.call().returns(PurseShape),
+
+    isLive: M.callWhen(M.await(PaymentShape)).returns(M.boolean()),
+    getAmountOf: M.callWhen(M.await(PaymentShape)).returns(amountShape),
+    burn: M.callWhen(M.await(PaymentShape))
+      .optional(M.pattern())
+      .returns(amountShape),
```

Comment:
If I understand correctly, the pattern for expressing the shape of a function is `M.{call,callWhen}(requiredArg1Shape, …).optional(optionalArg1Shape, …).returns(returnShape)`, where `callWhen` looks for shapes output by `M.await` as arguments that should be awaited before invoking the corresponding function. I accept it in this pull request, but there's probably room for more iteration. Some possibilities:
1. Merge `call` and `callWhen`, relying solely on `M.await` for what they currently differentiate.
2. Rename `call` to something like `callable`, better matching the general use of noun-named `M` methods. Or maybe not, depending upon number 4 below.
3. Make `M.await` infectious, such that `combine` can be updated to e.g. `M.callable(M.arrayOf(M.await(PaymentShape))).…` and lose its `Promise.all`. But this one might be tricky because `M.await` indicates call instructions rather than a shape assertion and therefore is really only appropriate for call/callWhen rather than in an arbitrary pattern—another alternative would be expanding that set such that `combine` can be updated to e.g. `M.callable(M.awaitAll(M.arrayOf(M.eref(PaymentShape)))).…`.
4. It is valuable for `M.await` to have a name indicating its distinction from the other methods that _are_ shape assertions. Noun vs. verb is at least a good start, but there are also stronger possibilities (e.g., substructure like `M.transform.{call,await,etc.}`).
5. The chainable/fluent interface appears distinct to `call`/`callWhen` (e.g., AFAIK we don't support something like `M.key().and(elementShape)`)—we should probably either expand it to the rest of `M` or drop it from these methods in favor of e.g. `M.call({ required: […], optional: […], returns: shape})`).

Reply:
> If I understand correctly, the pattern for expressing the shape of a function is `M.{call,callWhen}(requiredArg1Shape, …).optional(optionalArg1Shape, …).returns(returnShape)`, where `callWhen` looks for shapes output by `M.await` as arguments that should be awaited before invoking the corresponding function.

Yes.

> I accept it in this pull request, but there's probably room for more iteration. Some possibilities:
> 
> 1. Merge `call` and `callWhen`, relying solely on `M.await` for what they currently differentiate.

`combine` would do well by this rule, since it just happens not to have anything to await, but if it did, it could become a `callWhen`/`await` without losing correctness.

But I rejected that because `call` implies a guarantee of in-order delivery. Purse `deposit` relies on that guarantee, which is why it does not tolerate a promise argument. For cases like `combine`, if you wanted to express the absence of an in-order delivery guarantee, you could just use a `callWhen` that happened not to have an `await`. However, if we expected that to be common, we should optimize that case do it doesn't pay for an extra turn it doesn't need. Currently it would.

> 2. Rename `call` to something like `callable`, better matching the general use of noun-named `M` methods. Or maybe not, depending upon number 4 below.

Indeed, that's what I had in mind -- the distinction between guards and patterns. OTOH, `M.interface` reads like a noun, which breaks my story.

> 3. Make `M.await` infectious, such that `combine` can be updated to e.g. `M.callable(M.arrayOf(M.await(PaymentShape))).…` and lose its `Promise.all`. But this one might be tricky because `M.await` indicates call instructions rather than a shape assertion and therefore is really only appropriate for call/callWhen rather than in an arbitrary pattern—another alternative would be expanding that set such that `combine` can be updated to e.g. `M.callable(M.awaitAll(M.arrayOf(M.eref(PaymentShape)))).…`.

I like the `M.awaitAll` idea. Altogether, I doubt the case is common enough to provide support for. The `combine` code shows that it is convenient enough to do it manually if it only arises rarely. If we do eventually decide that we want better support, I we should do `M.awaitAll`.

> 4. It is valuable for `M.await` to have a name indicating its distinction from the other methods that _are_ shape assertions. Noun vs. verb is at least a good start, but there are also stronger possibilities (e.g., substructure like `M.transform.{call,await,etc.}`).
>
> 5. The chainable/fluent interface appears distinct to `call`/`callWhen` (e.g., AFAIK we don't support something like `M.key().and(elementShape)`)—we should probably either expand it to the rest of `M` or drop it from these methods in favor of e.g. `M.call({ required: […], optional: […], returns: shape})`).

For both 4 and 5, I'm trying for a sweet spot on brevity and readability as well. Many more functions have required arguments than have optional or rest arguments.

IIUC, all your suggestions above are tentative, such that you're ok with me merging without having changed the code itself in light of these suggestions, given my responses. But please keep these ideas in mind for possible improvements after this PR. Resolving.